### PR TITLE
Preparing Shared for push alarm not. - SLIM-762

### PR DESCRIPTION
**Pull request by hannek24**

### CODE REVIEW

File: osgp-dto/src/main/java/com/alliander/osgp/dto/valueobjects/smartmetering/PushNotificationAlarmDto.java [hunk 1]
```diff
@@ -17,9 +18,14 @@
 
     private final String deviceIdentification;
     private final EnumSet<AlarmTypeDto> alarms;
+    private final byte[] bytes;
```
Comment (rudidevries) [inline osgp-dto/src/main/java/com/alliander/osgp/dto/valueobjects/smartmetering/PushNotificationAlarmDto.java]: For this field and all references to it, I don't think bytes is the right name. It says what the datatype is, is should say what the data represents.

File: osgp-ws-smartmetering/src/main/resources/schemas/sm-configuration.xsd [hunk 2]
```diff
@@ -107,6 +107,54 @@
...
+      </xsd:complexContent>
+    </xsd:complexType>
+  </xsd:element>
+  <xsd:element name="GetPushNotificationAlarmResponse">
+    <xsd:complexType>
+      <xsd:sequence>
+        <xsd:element name="Result" type="common:OsgpResultType"
+          minOccurs="1" />
+        <xsd:element name="decodedMessage" type="xsd:string" />
+        <xsd:element name="encodedMessage" type="xsd:string" />
```
Comment (rudidevries) [inline osgp-ws-smartmetering/src/main/resources/schemas/sm-configuration.xsd]: Looking at the scenario the encoded message contains hex byte data. So this field should be of type xsd:hexBinary.

File: osgp-ws-smartmetering/src/main/resources/schemas/sm-configuration.xsd [hunk 3]
```diff
@@ -107,6 +107,54 @@
...
+        <xsd:sequence>
+          <xsd:element name="alarm" type="tns:AlarmType"
+            minOccurs="1" maxOccurs="unbounded" />
+        </xsd:sequence>
+      </xsd:sequence>
+    </xsd:complexType>
+  </xsd:element>
+
+  <xsd:simpleType name="AlarmType">
```
Comment (rudidevries) [inline osgp-ws-smartmetering/src/main/resources/schemas/sm-configuration.xsd]: Alarmtypes are defined in commen, so should be included an not copied.

File: osgp-ws-smartmetering/src/main/resources/schemas/sm-configuration.xsd [hunk 4]
```diff
@@ -107,6 +107,54 @@
     </xsd:complexType>
   </xsd:element>
 
+  <xsd:element name="GetPushNotificationAlarmRequest">
```
Comment (rudidevries) [inline osgp-ws-smartmetering/src/main/resources/schemas/sm-configuration.xsd]: Following the naming convention this should be named GetPushNotificationAlarmAsyncRequest.

File: osgp-ws-smartmetering/src/main/resources/SmartMeteringConfiguration.wsdl [hunk 5]
```diff
@@ -48,6 +48,17 @@
     </wsdl:part>
   </wsdl:message>
 
+  <wsdl:message name="GetPushNotificationAlarmRequest">
+    <wsdl:part element="smconf:GetPushNotificationAlarmRequest"
```
Comment (rudidevries) [inline osgp-ws-smartmetering/src/main/resources/SmartMeteringConfiguration.wsdl]: Naming following the convention should indicate that it is a AsynRequest.

File: osgp-ws-smartmetering/src/main/resources/SmartMeteringConfiguration.wsdl [hunk 6]
```diff
@@ -286,6 +297,13 @@
       </wsdl:output>
     </wsdl:operation>
 
+    <wsdl:operation name="GetPushNotificationAlarm">
+      <wsdl:input message="tns:GetPushNotificationAlarmRequest" name="GetPushNotificationAlarmRequest">
```
Comment (rudidevries) [inline osgp-ws-smartmetering/src/main/resources/SmartMeteringConfiguration.wsdl]: Naming following the convention should indicate that it is a AsynRequest.

File: osgp-ws-smartmetering/src/main/resources/schemas/sm-configuration.xsd (hunk 4 above)
Comment (hannek24) [inline osgp-ws-smartmetering/src/main/resources/schemas/sm-configuration.xsd]: It's not async. Async has nothing to do with it. Here it's classical request-response.

File: osgp-ws-smartmetering/src/main/resources/SmartMeteringConfiguration.wsdl (hunk 5 above)
Comment (hannek24) [inline osgp-ws-smartmetering/src/main/resources/SmartMeteringConfiguration.wsdl]: Again it's request-response. Async is not applicable.

**Pull request merged**